src/Features/CSharp/Portable/CodeRefactorings/PullMemberUp/CSharpPullMemberUpCodeRefactoringProvider.cs

```diff
@@ -35,6 +52,35 @@ protected override bool IsSelectionValid(TextSpan span, SyntaxNode selectedNode)
             }
         }
 
+        protected async override Task<SyntaxNode> GetMatchedNodeAsync(Document document, TextSpan span)
```

Comment:
@CyrusNajmabadi 
This is a new method I added in order to get the right SyntaxNode at the beginning.
The original problem is when the cursor stands between a Method identifier and parenthesis, root.FindNode() will not return the SyntaxNode contains the identifier.

Reply:
Also, if you use that helper i mentioned, we should be able to do this in the base class.

Reply:
Change it using GetTouchingWordAsync()